core/src/test/java/org/elasticsearch/action/support/replication/ShardReplicationTests.java

```diff
@@ -314,11 +352,10 @@ public void testRoutingToPrimary() {
                 assertIndexShardCounter(1);
             }
         } else {
-            logger.info("--> primary is assigned to [{}], checking request forwarded", primaryNodeId);
-            final List<CapturingTransport.CapturedRequest> capturedRequests = transport.capturedRequestsByTargetNode().get(primaryNodeId);
-            assertThat(capturedRequests, notNullValue());
-            assertThat(capturedRequests.size(), equalTo(1));
-            assertThat(capturedRequests.get(0).action, equalTo("testAction"));
+            logger.info("--> primary is assigned to [{}], checking request is going to be retried at some point", primaryNodeId);
+            assertThat(clusterService.getListeners().size(), equalTo(0));
```

Comment:
I don't understand why we would expect a retry here. Cluster state is fine, this should just send the request.  Also, why remove the check for captured requests?